cfme/fixtures/v2v.py

```diff
@@ -163,32 +165,28 @@ def get_vm(request, appliance, source_provider, template, datastore="nfs"):
     return vm_obj
 
 
-def _form_data(source_provider, provider):
-    form_data = {
-        "general": {
-            "name": "infra_map_{}".format(fauxfactory.gen_alphanumeric()),
-            "description": "Single Datastore migration of VM from {ds_type1} to"
-            " {ds_type2}".format(ds_type1="nfs", ds_type2="nfs"),
-        },
-        "cluster": {"mappings": [_form_data_mapping("clusters", source_provider, provider)]},
-        "datastore": {
-            "Cluster ({})".format(provider.data.get("clusters")[0]): {
-                "mappings": [
-                    _form_data_mapping("datastores", source_provider, provider, "nfs", "nfs")
-                ]
-            }
-        },
-        "network": {
-            "Cluster ({})".format(provider.data.get("clusters")[0]): {
-                "mappings": [
-                    _form_data_mapping(
-                        "vlans", source_provider, provider, "VM Network", "ovirtmgmt"
-                    )
-                ]
-            }
-        },
+def infra_mapping_default_data(source_provider, provider):
+    """
+    Default data for infrastructure mapping form.
+    It is used in other methods to recursive update the data accoriding
+    to parameters in tests.
+
+    Args:
+        source_provider: Vmware provider
+        provider: Target rhev/OSP provider
+    """
+    infra_mapping_data = {
+        "name": "infra_map_{}".format(fauxfactory.gen_alphanumeric()),
+        "description": "Single Datastore migration of VM from {ds_type1} to {ds_type2}".format(
+            ds_type1="nfs", ds_type2="nfs"
+        ),
+        "clusters": [component_generator("clusters", source_provider, provider)],
+        "datastores": [component_generator("datastores", source_provider, provider, "nfs", "nfs")],
+        "networks": [
+            component_generator("vlans", source_provider, provider, "VM Network", "ovirtmgmt")
```

Comment:
It looks like you're picking specific values here to pass, and hard coding them into the test. `VM Network`  and `ovirtmgmt` are network/vlan names on the providers themselves, and should be pulled from yaml, not hardcoded.

Reply:
This is just to create a data structure for default data . 
Not used in tests directly .
Only used after doing recursive updates from request.params

Reply:
This is not true, as `mapping_data_multiple_vm_obj_single_datastore` does not modify clusters or datastores, and `mapping_data_single_datastore` does not modify clusters or networks.

If we're going to have some default values here, we need to justify where they are coming from. Are they default network names for VMware and RHEVM? If so we at least need to state them here - but again, why not pull them from the provider provisioning data? You have both provider objects available.